### PR TITLE
Add `background_opacity` argument to `add_legend`

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add `background_opacity` argument to `add_legend`.

### Details

- None

### CODE REVIEW

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

File: tests/plotting/test_renderer.py [hunk 1]
```diff
@@ -177,3 +178,11 @@ def test_legend_face(sphere, face):
     pl = pv.Plotter()
     pl.add_mesh(sphere, label='sphere')
     pl.add_legend(face=face)
+
+
+def test_add_legend_background_opacity(sphere):
+    background_opacity = default_rng(0).random()
```
Comment (MatthewFlamm) [inline tests/plotting/test_renderer.py:184]: Small point: Is there a reason not to hard code a value here?  Seeding a RNG will result in the same value each time as well, we might as well make it explicit for easier human understanding.

File: pyvista/plotting/renderer.py [hunk 2]
```diff
@@ -3357,6 +3358,9 @@ def add_legend(
             or ``'arial'``. Defaults to :attr:`pyvista.global_theme.font.family
             <pyvista.plotting.themes._Font.family>`.
 
+        background_opacity : int | float, default: 1.0
```
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:3361]: I'm assuming that it should be a float input, but an int can be promoted to a float in python/vtk?

```suggestion
        background_opacity : float, default: 1.0
```

**Comment by MatthewFlamm** [comment]
#3015 is still being kept current.  I haven't checked.  But if there will be a conflict, it might be better to hold off here until that one is merged?

File: pyvista/plotting/renderer.py (hunk 2 above)
Comment (tkoyama010) [reply pyvista/plotting/renderer.py:3361]: Good catch. Thanks.

File: tests/plotting/test_renderer.py (hunk 1 above)
Comment (tkoyama010) [reply tests/plotting/test_renderer.py:184]: Only my mistake. I want to test each time with a random value between 0 and 1. Thank you for catching it.

```suggestion
    background_opacity = default_rng().random()
```

Comment (tkoyama010) [reply tests/plotting/test_renderer.py]: I see, that is easier to understand.

File: tests/plotting/test_renderer.py [hunk 3]
```diff
@@ -177,3 +178,11 @@ def test_legend_face(sphere, face):
     pl = pv.Plotter()
     pl.add_mesh(sphere, label='sphere')
     pl.add_legend(face=face)
+
+
+def test_add_legend_background_opacity(sphere):
+    background_opacity = default_rng().random()
```
Comment (tkoyama010) [inline tests/plotting/test_renderer.py:184]:
```suggestion
    background_opacity = 0.4
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: tests/plotting/test_renderer.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+from numpy.random import default_rng
```
Comment (MatthewFlamm) [inline tests/plotting/test_renderer.py]: Last bit

```suggestion
```

Comment (tkoyama010) [reply tests/plotting/test_renderer.py]: Thanks!

**Review by MatthewFlamm — APPROVED** [review]
LGTM but worth looking at #3015 before merging.

**Comment by tkoyama010** [comment]
Thanks! I will address the conflicts in #5509, as changing by fixing the conflicts here would require another approval.

**Pull request merged**